### PR TITLE
Fetch in react - Laura Grace Manimtim

### PR DESCRIPTION
**Pull request by lauragracem**
Screen Shot 2022-07-20 at 12 26 22 PM

### CODE REVIEW

File: fetch-in-react/src/app.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    fetch('https://jsonplaceholder.typicode.com/users')
+      .then(response => response.json())
+      .then(users => {
+        this.setState({ users });
+        this.setState({ isLoading: false });
```
Comment (scbowler) [inline fetch-in-react/src/app.jsx:18]: This should be done with a single `setState` call:
```JAVASCRIPT
this.setState({ users, isLoading: false });
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just one update needed. Let me know if you have any questions.

File: fetch-in-react/src/app.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      users: [],
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    fetch('https://jsonplaceholder.typicode.com/users')
+      .then(response => response.json())
+      .then(users => {
+        this.setState({ isLoading: false });
```
Comment (scbowler) [inline fetch-in-react/src/app.jsx:17]: You need to update the `users` in state and `isLoading`. Make sure you are testing your code when you make updates.
```JAVASCRIPT
this.setState({ users: users, isLoading: false });
```

**Review by scbowler — CHANGES_REQUESTED** [review]
One more update needed. Let me know if you have any questions.

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**